### PR TITLE
Add login/logout functionality

### PR DESCRIPTION
**Pull request by habdelra**
Also lightly edit the user profile to prove access token works

https://user-images.githubusercontent.com/61075/236502448-78e966f6-4072-4d23-a45c-a8fa1c13f111.mp4

### CODE REVIEW

**Review by backspace — APPROVED** [review]
I tried this locally, total success 🎉

File: packages/host/app/components/login.gts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  BoxelInput,
+  Button,
+  FieldContainer,
+  LoadingIndicator,
+} from '@cardstack/boxel-ui';
+import { isMatrixError } from '../lib/matrix-utils';
+import type MatrixService from '../services/matrix-service';
+import { type IAuthData } from 'matrix-js-sdk';
+
+const TRUE = true;
```
Comment (jurgenwerk) [inline packages/host/app/components/login.gts:18]: How come a constant is needed for `true`?

**Review by jurgenwerk — APPROVED** [review]
Looks great

File: packages/host/app/components/login.gts (hunk 1 above)
Comment (habdelra) [reply packages/host/app/components/login.gts:18]: The prettier for glint turns a `true` in the template to `”true”`—very annoying.

**Pull request merged**